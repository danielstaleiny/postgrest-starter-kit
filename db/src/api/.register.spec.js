const ROOT = process.cwd()
const { get, post, put, patch, del } = require(`${ROOT}/util/fetch.js`)(
  'http://localhost:3000/'
)

describe('/rpc/register', () => {
  test('Success 201', async () => {
    const res = await post('rpc/register')({
      body: {
        name: 'New user',
        email: 'test@email.com',
        password: 'pass',
      },
    })
    res.body.token = 'checked'
    res.body.refresh_token = 'checked'
    res.body.csrf = 'checked'
    expect(res).toMatchSnapshot()
  })

  test('Success 201, no csrf', async () => {
    const res = await post('rpc/register')({
      body: {
        name: 'New user2',
        email: 'test2@email.com',
        password: 'pass',
        csrf: false,
      },
    })
    res.body.token = 'checked'
    res.body.refresh_token = 'checked'
    expect(res).toMatchSnapshot()
  })
})
