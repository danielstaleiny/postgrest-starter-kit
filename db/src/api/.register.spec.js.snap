// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/rpc/register Success 201 1`] = `
Object {
  "body": Object {
    "csrf": "checked",
    "email": "test@email.com",
    "id": 3,
    "name": "New user",
    "refresh_token": "checked",
    "role": "webuser",
    "token": "checked",
  },
  "status": 200,
  "statusText": "OK",
}
`;

exports[`/rpc/register Success 201, no csrf 1`] = `
Object {
  "body": Object {
    "email": "test2@email.com",
    "id": 4,
    "name": "New user2",
    "refresh_token": "checked",
    "role": "webuser",
    "token": "checked",
  },
  "status": 200,
  "statusText": "OK",
}
`;
